src/Cardano/Wallet/Api/JSON.hs

```diff
@@ -0,0 +1,26 @@
+module Cardano.Wallet.Api.JSON
+    ( defaultSumTypeOptions
+    , defaultRecordTypeOptions
+    ) where
+
+import Prelude
+
+import Data.Aeson
+    ( Options
+    , camelTo2
+    , constructorTagModifier
+    , defaultOptions
+    , fieldLabelModifier
+    , omitNothingFields
+    , tagSingleConstructors
+    )
+
+defaultSumTypeOptions :: Options
+defaultSumTypeOptions = defaultOptions
+    { constructorTagModifier = camelTo2 '_'
+    , tagSingleConstructors = True }
+
+defaultRecordTypeOptions :: Options
+defaultRecordTypeOptions = defaultOptions
+    { fieldLabelModifier = drop 1
```

Comment:
I guess we also need to transform from `camelCase` to `snake_case` for all fields as well!

Reply:
Done!